components/ledger/internal/services/command/create_balance_transaction_operations_async.go

```
@@ -239,6 +265,40 @@ func (uc *UseCase) RemoveTransactionFromRedisQueue(ctx context.Context, logger l
+		logger.Log(ctx, libLog.LevelWarn, fmt.Sprintf("Backup queue: failed to decode transaction %s before conditional cleanup: %v", transactionKey, err))
+		return
+	}
+
+	if !strings.EqualFold(queue.TransactionStatus, expectedStatus) {
+		logger.Log(ctx, libLog.LevelInfo, fmt.Sprintf(
+			"Backup queue: skip cleanup for transaction %s because status changed (expected=%s current=%s)",
+			transactionKey,
+			expectedStatus,
+			queue.TransactionStatus,
+		))
+		return
```

golangci-lint (2.11.4)

[error] missing whitespace above this line (too many lines above return)

(wsl_v5)